### PR TITLE
cmake: Incorporate gtest in a standard way

### PR DESCRIPTION
**Pull request by MOON-CLJ**
following this guide: https://github.com/google/googletest/blob/6d0bb75c81bc59770520e303f6a04737b8aedb39/googletest/README.md#incorporating-into-an-existing-cmake-project

### CODE REVIEW

File: src/Client.cpp [hunk 1]
```diff
@@ -30,7 +29,6 @@ void Client::config(config_options_t opt, int val) {
       break;
     case CFG_HASH_FUNCTION:
       ConnectionPool::setHashFunction(static_cast<hash_function_options_t>(val));
-    default:
```
Comment (windreamer) [inline src/Client.cpp]: 如果没记错， `default` 没有会有warning？

Comment (MOON-CLJ) [reply src/Client.cpp]: @windreamer https://travis-ci.org/douban/libmc/jobs/315153298 没看到这行的warning。cppcheck我下个pr修，我本地ubuntu14.04更多warning 。

**Comment by MOON-CLJ** [comment]
@youngsofun pls review cc @windreamer @mckelvin

File: include/Common.h [hunk 2]
```diff
@@ -172,29 +172,45 @@ typedef enum {
 
 
 typedef enum {
-  // storage commands -> key_va -> success_or_failure
```
Comment (mckelvin) [inline include/Common.h]: Would you please move these non-gtest related code into a separate commit?

**Comment by MOON-CLJ** [comment]
@mckelvin 

> it will still download and unpack gtest when running cmake ...

seems not

```
clj@vagrant ~/d/libmc> mkdir build
clj@vagrant ~/d/libmc> cd build/
clj@vagrant ~/d/l/build> cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_TESTING=ON  ..
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Git: /usr/bin/git (found version "1.9.1")
-- Configuring done
-- Generating done
-- Build files have been written to: /home/clj/dev/libmc/build
clj@vagrant ~/d/l/build> ls ext/gtest/src/
googletest/  googletest-build/  googletest-stamp/
clj@vagrant ~/d/l/build> ls ext/gtest/src/googletest
clj@vagrant ~/d/l/build> ls ext/gtest/src/googletest-build/
```

File: include/Common.h (hunk 2 above)
Comment (MOON-CLJ) [reply include/Common.h]: okay

Comment (MOON-CLJ) [reply include/Common.h]: @mckelvin non-gtest related code will put in next pr。

**Comment by mckelvin** [comment]
@MOON-CLJ You right, the gtest will only be built when `make` is called. But doesn't that matter? 

I guess you don't care whether it's built during `cmake`-time or `make`-time. The original purpose of the PR seems to integrate cmake with gtest in a standard way, which is good. If that's true, I think the title of the PR should be renamed something like "cmake: Incorporate gtest in a standard way"

**Comment by MOON-CLJ** [comment]
@mckelvin  "cmake: Incorporate gtest in a standard way" lgtm，that's what i want to do。the msg came from https://github.com/douban/libmc/pull/58/files#diff-af3b638bc2a3e6c650974192a53c7291R51 :)。

**Comment by MOON-CLJ** [comment]
i'm dealing with the ci error。

**Comment by MOON-CLJ** [comment]
i think one of the matrix ci failed, just because some small probability event about network，so  i try https://github.com/douban/libmc/pull/59 to make the err more clear。i will rebase it。

**Pull request merged**